### PR TITLE
[CMake] add support for find_package

### PR DESCRIPTION
**Pull request by prateek9623**
This PR adds the following features:
- add target tvm to a CMake export group, which enables its linkage with another target with another target group.
- adds support for cmake find_package

### CODE REVIEW

**Comment by jroesch** [comment]
cc @tkonolige can you take a look at this one?

File: CMakeLists.txt [hunk 1]
```diff
@@ -620,33 +611,54 @@ endif()
...
     PATTERN "*.h"
     )
 endif(INSTALL_DEV)
 
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+set(PROJECT_CONFIG_CONTENT "@PACKAGE_INIT@\n")
+string(APPEND PROJECT_CONFIG_CONTENT
+       "include(\"\${CMAKE_CURRENT_LIST_DIR}/${PROJECT_NAME}Targets.cmake\")")
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/PROJECT_CONFIG_FILE" ${PROJECT_CONFIG_CONTENT})
```
Comment (tkonolige) [inline CMakeLists.txt:646]:
```suggestion
file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/tvm_config_file.cmake" ${PROJECT_CONFIG_CONTENT})
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -620,33 +611,54 @@ endif()
...
+
+install(EXPORT ${PROJECT_NAME}Targets
+  NAMESPACE ${PROJECT_NAME}::
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+
+# Create config for find_package()
+configure_package_config_file(
+  "${CMAKE_CURRENT_BINARY_DIR}/PROJECT_CONFIG_FILE" ${PROJECT_NAME}Config.cmake
```
Comment (tkonolige) [inline CMakeLists.txt:654]:
```suggestion
  "${CMAKE_CURRENT_BINARY_DIR}/tvm_config_file.cmake" ${PROJECT_NAME}Config.cmake
```

**Review by tkonolige — CHANGES_REQUESTED** [review]
Thank you @prateek9623 for this PR. It really helps improve the usability of tvm by other projects.

One issue though: the exported config file needs to `find_package(Threads REQUIRED)` as it is a `PUBLIC` dependency of tvm.

File: CMakeLists.txt (hunk 1 above)
Comment (prateek9623) [reply CMakeLists.txt:648]: this to generate temp file, final file name will be "tvmConfig.cmake"

File: CMakeLists.txt (hunk 2 above)
Comment (prateek9623) [reply CMakeLists.txt:656]: this to generate temp file, final file name will be "tvmConfig.cmake"

**Comment by prateek9623** [comment]
@tkonolige I have Threads as the public dependency, and as per PROJECT_CONFIG_FILE it is just to create a temporary file, its content is just moved in line no: https://github.com/apache/tvm/blob/dc0550ec04f40817c7472311f531728941199a25/CMakeLists.txt#L655

File: CMakeLists.txt (hunk 2 above)
Comment (tkonolige) [reply CMakeLists.txt:656]: How about `temp_config_file.cmake` then?

**Comment by prateek9623** [comment]
@tkonolige done with requested changes

**Review by jroesch — APPROVED** [review]

**Pull request merged**